ext/strscan/strscan.c

```diff
@@ -1669,17 +1669,15 @@ name_to_backref_number(struct re_registers *regs, VALUE regexp, const char* name
 {
     int num;
 
-    num = onig_name_to_backref_number(RREGEXP_PTR(regexp),
-	(const unsigned char* )name, (const unsigned char* )name_end, regs);
-    if (num >= 1) {
-	return num;
-    }
-    else {
-	rb_enc_raise(enc, rb_eIndexError, "undefined group name reference: %.*s",
-					  rb_long2int(name_end - name), name);
+    if (RTEST(regexp)){
+        num = onig_name_to_backref_number(RREGEXP_PTR(regexp),
```

Comment:
Can we narrow `num` scope?

```suggestion
        int num = onig_name_to_backref_number(RREGEXP_PTR(regexp),
```

Reply:
done